src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -2298,8 +2296,7 @@ def attach_managed_data_disk_to_vmss(cmd, resource_group_name, vmss_name, size_g
     def _init_data_disk(storage_profile, lun, existing_disk=None):
         data_disks = storage_profile.data_disks or []
         if lun is None:
-            luns = [d.lun for d in data_disks]
-            lun = max(luns) + 1 if luns else 0
+            lun = _get_disk_lun(data_disks)
         if existing_disk is None:
             data_disk = DataDisk(lun=lun, create_option=DiskCreateOptionTypes.empty, disk_size_gb=size_gb,
                                  caching=caching, managed_disk=ManagedDiskParameters(storage_account_type=sku))
```

Comment:
I see that in line 2305 we can attach an existing data disk if the ID is specified and `--new` is not included in the command.

Should we add some validation to make sure that the `disk` variable we are passing to create a `DataDisk` is a valid resource ID, if --new is not specified? I imagine we should add some check too for the vmss version of this method, as the help for `az vmss disk attach` says `--disk` is a neme or ID

i just tried, and the command fails. If we add some validation, we can give a clearer error, or just assume the user wants to make a new disk.

```
(cli-venv) Oluwatosins-MBP:azure-cli tosin$ az vm disk attach --vm-name ovaTestVM --disk disk1
The entity was not found.
```

Reply:
This error should be fixed by service end instead of letting client polish it. Because no users have ever complained about this error, i would just live with it.

Reply:
Thanks for showing me the validator on the argument.